services/credentials-service/src/revocation-list/revocation-list.service.ts

```diff
@@ -0,0 +1,259 @@
+import { HttpService } from '@nestjs/axios';
+import { Injectable, InternalServerErrorException, Logger } from '@nestjs/common';
+import { CredentialPayload, transformCredentialInput } from 'did-jwt-vc';
+import { DIDDocument } from 'did-resolver';
+import { IdentityUtilsService } from '../credentials/utils/identity.utils.service';
+// import { PrismaService } from 'src/prisma.service';
+import { PrismaClient } from '@prisma/client';
+import { IssuerType, Proof } from 'did-jwt-vc/lib/types';
+import { JwtCredentialSubject } from 'src/app.interface';
+import { RevocationLists, VerifiableCredentials } from '@prisma/client';
+import { RevocationListImpl } from './revocation-list.impl';
+import { RevocationList } from './revocation-list.helper';
+
+@Injectable()
+export class RevocationListService {
+  constructor(
+    private readonly prismaService: PrismaClient,
+    private readonly rl: RevocationListImpl,
+    private readonly identityService: IdentityUtilsService) {}
+
+  private async signRevocationListCredential(revocationListCredential, issuer) {
+    try {
+      revocationListCredential['proof'] = {
+        proofValue: await this.identityService.signVC(
+          transformCredentialInput(revocationListCredential as CredentialPayload),
+          issuer,
+        ),
+        type: 'Ed25519Signature2020',
+        created: new Date().toISOString(),
+        verificationMethod: issuer,
+      };
+    } catch (err) {
+      Logger.error('Error signing revocation list', err);
+      throw new InternalServerErrorException(
+        'Error signing revocation list',
+      );
+    }
+
+    return revocationListCredential.proof;
+  }
+
+  private generateRevocationListCredentialSkeleton(id: string, issuer: string) {
+    return {
+      "@context": [
+        "https://www.w3.org/2018/credentials/v1",
+        "https://w3id.org/vc-revocation-list-2020/v1"
+      ],
+      id,
+      type: ["VerifiableCredential", "RevocationList2020Credential"],
+      issuer,
+      issuanceDate: new Date().toISOString(),
+      credentialSubject: {
+        id,
+        type: "RevocationList2020",
+      },
+      proof: {}
+    };
+  }
+  async createNewRevocationList(issuer: string) {
+    // generate did for the revocation list
+    let credDID: ReadonlyArray<DIDDocument>;
+    try {
+      credDID = await this.identityService.generateDID(['verifiable credential']);
+    } catch (err) {
+      Logger.error('Error generating DID for revocation list', err);
+      throw new InternalServerErrorException(
```

Comment:
Instead of InternalServerErrorException, can we create an application specific exception.